nxc/modules/powershell_history.py

```diff
@@ -0,0 +1,84 @@
+import traceback
+import os
+from impacket.examples.secretsdump import RemoteOperations
+
+class NXCModule:
+    """Module by @357384n"""
+
+    name = "powershell_history"
+    description = "Extracts PowerShell history for all users and looks for sensitive commands."
+    supported_protocols = ["smb"]
+    opsec_safe = True
+    multiple_hosts = True
+
+    def options(self, context, module_options):
+        """To export all the history you can add the following option: -o export=enable"""
+        context.log.info(f"Received module options: {module_options}")
+        self.export = module_options.get('EXPORT', 'disable').lower()
+        context.log.info(f"Option export set to: {self.export}")
+
+    def execute_command(self, connection, command):
+        """Execute a command on the remote system and return the output."""
+        output = connection.execute(command, True)
+        return output
+
+    def get_powershell_history(self, connection):
+        """Get the PowerShell history for all users."""
+        history_paths_command = 'powershell.exe "type C:\\Users\\*\\AppData\\Roaming\\Microsoft\\Windows\\PowerShell\\PSReadLine\\ConsoleHost_history.txt"'
+        try:
+            history_output = self.execute_command(connection, history_paths_command)
+            return history_output.split('\n')
+        except Exception as e:
+            raise Exception(f"Could not retrieve PowerShell history: {e}")
```

Comment:
Ruff complains about it as well, but this should either be removed or not raise an exception. The later probably doesn't make sense since we can't really continue without the command output.